### ISSUE TITLE
Crash on pretty much any action

### ISSUE THREAD

**Issue by MrConorAE** [t1m0thyj/WinDynamicDesktop#324]
- WDD Version: 4.5.0
- OS Version: Windows 10 Home 21H1 (Build 19043.1023)

Steps to Reproduce:

1. Do almost anything with the app (start it up [this stopped with a fresh install], try downloading a theme, try applying a theme). Opening "About" does NOT trigger the error.

Are you using the Microsoft Store version of WDD?: Yes

Log of error:
```
[2021-06-01 20:51:49.824] System.IndexOutOfRangeException: Der Index war außerhalb des Arraybereichs.
   bei WinDynamicDesktop.WallpaperChangeScheduler.GetImageData(SolarData data, ThemeConfig theme, DateTime dateNow)
   bei WinDynamicDesktop.WPF.ThemePreviewerViewModel.PreviewTheme(ThemeConfig theme)
   bei WinDynamicDesktop.ThemeDialog.UpdateSelectedItem()
   bei WinDynamicDesktop.ThemeDialog.OnDownloadDialogClosed(Object sender, FormClosedEventArgs e)
   bei System.Windows.Forms.Form.OnFormClosed(FormClosedEventArgs e)
   bei System.Windows.Forms.Form.WmClose(Message& m)
   bei System.Windows.Forms.Form.WndProc(Message& m)
   bei System.Windows.Forms.Control.ControlNativeWindow.OnMessage(Message& m)
   bei System.Windows.Forms.Control.ControlNativeWindow.WndProc(Message& m)
   bei System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)
```

`Der Index war außerhalb des Arraybereichs.` is `The index is out of range`.

**Comment by MrConorAE** [comment]
FYI: this also happens on the GitHub version of the app.

**Comment by t1m0thyj** [comment]
@MrConorAE Can you upload the debug.log file here? It's in the WDD folder.

**Comment by MrConorAE** [comment]
[debug.log](https://github.com/t1m0thyj/WinDynamicDesktop/files/6580570/debug.log)
Here you go :)

**Comment by MrConorAE** [comment]
Any idea what's going on? I'm not sure if it's just a problem my end, or if it's WDD.

**Comment by t1m0thyj** [comment]
From the log, I can see WDD detects you're in/near New Zealand, but I also noticed from the error message that you're using the German translation. This makes me curious if your timezone is in Europe or somewhere else outside of NZ? I'm wondering if WDD has an issue when your actual location and your system timezone's location are on opposite sides of the International Date Line.

**Comment by MrConorAE** [comment]
Yeah, it's a bit of an odd situation.
I live in New Zealand, my timezone is New Zealand and my "region" is set to New Zealand...
... but I'm currently learning German, so I've set my devices to it to help myself learn :D
Do you think that "clash" of timezone and locale is causing it?
-------- Original-Nachricht --------

**Comment by t1m0thyj** [comment]
> Do you think that "clash" of timezone and locale is causing it?

I was thinking so, but just tested setting my system timezone to NZ in a Win10 VM and that was enough to crash WDD without changing the locale. So I think there is a bug for NZ 😢 I'll look into this more.

**Comment by t1m0thyj** [comment]
@MrConorAE I think I've fixed the issue, could you test this build of WDD? https://ci.appveyor.com/project/t1m0thyj/windynamicdesktop/builds/39477140/artifacts

**Comment by Excigma** [comment]
Same issue here, also in NZ
   ```log
[2021-06-04 05:17:39.593] System.IndexOutOfRangeException: Index was outside the bounds of the array.
   at WinDynamicDesktop.WallpaperChangeScheduler.GetImageData(SolarData data, ThemeConfig theme, DateTime dateNow)
   at WinDynamicDesktop.WallpaperChangeScheduler.RunScheduler(Boolean forceImageUpdate)
   at WinDynamicDesktop.AppContext..ctor(String[] args)
   at WinDynamicDesktop.Program.Main(String[] args)```

   However the test build works with no issues

**Comment by MrConorAE** [comment]
Which build should I use? Portable, installer or UWP?

**Comment by t1m0thyj** [comment]
Whichever you would regularly use, they're all the same build 🙂 
Portable and Installer are the same files that get uploaded to GitHub Releases, UWP = Store app

**Comment by MrConorAE** [comment]
It works! Thank you _so_ much for fixing this, I missed it :)

**Comment by t1m0thyj** [comment]
Fix has been released in WDD 4.6.0
